core/includes/filetransfer/filetransfer.inc

```diff
@@ -61,19 +106,60 @@ abstract class FileTransfer {
    * If the connection isn't set to anything, this will call the connect() method
    * and set it to and return the result; afterwards, the connection will be
    * returned directly without using this method.
+   *
+   * @param string $name
+   *   The name of the variable to return.
+   *
+   * @return string|bool
+   *   The variable specified in $name.
    */
   function __get($name) {
     if ($name == 'connection') {
       $this->connect();
-      return $this->connection;
+      return $this->connectionHandle;
     }
 
     if ($name == 'chroot') {
       $this->setChroot();
-      return $this->chroot;
+      return $this->chrootPath;
+    }
+  }
+
+  /**
+   * {@inheritdoc}
```

Comment:
`{@inheritdoc}` is used when there's a parent class that has this same method. But this `abstract class FileTransfer` does not extend any other classes. Could you document this and the other new magic methods?